pkg/controllers/nodepool/registrationhealth/suite_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registrationhealth_test
+
+import (
+	"context"
+	"testing"
+
+	"sigs.k8s.io/karpenter/pkg/controllers/nodepool/registrationhealth"
+
+	"github.com/awslabs/operatorpkg/object"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/karpenter/pkg/apis"
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider/fake"
+	"sigs.k8s.io/karpenter/pkg/test"
+	. "sigs.k8s.io/karpenter/pkg/test/expectations"
+	"sigs.k8s.io/karpenter/pkg/test/v1alpha1"
+	. "sigs.k8s.io/karpenter/pkg/utils/testing"
+)
+
+var (
+	controller    *registrationhealth.Controller
+	ctx           context.Context
+	env           *test.Environment
+	cloudProvider *fake.CloudProvider
+	nodePool      *v1.NodePool
+	nodeClass     *v1alpha1.TestNodeClass
+)
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "RegistrationHealth")
+}
+
+var _ = BeforeSuite(func() {
+	cloudProvider = fake.NewCloudProvider()
+	env = test.NewEnvironment(test.WithCRDs(apis.CRDs...), test.WithCRDs(v1alpha1.CRDs...))
+	controller = registrationhealth.NewController(env.Client, cloudProvider)
+})
+var _ = AfterEach(func() {
+	ExpectCleanedUp(ctx, env.Client)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = Describe("RegistrationHealth", func() {
+	BeforeEach(func() {
+		nodePool = test.NodePool()
+		nodeClass = test.NodeClass(v1alpha1.TestNodeClass{
+			ObjectMeta: metav1.ObjectMeta{Name: nodePool.Spec.Template.Spec.NodeClassRef.Name},
+		})
+		nodePool.Spec.Template.Spec.NodeClassRef.Group = object.GVK(nodeClass).Group
+		nodePool.Spec.Template.Spec.NodeClassRef.Kind = object.GVK(nodeClass).Kind
+		_ = nodePool.StatusConditions().Clear(v1.ConditionTypeNodeRegistrationHealthy)
+	})
+	It("should ignore setting NodeRegistrationHealthy status condition on NodePools which aren't managed by this instance of Karpenter", func() {
```

Comment:
How about changing the reason to `the referenced nodeClass is not supported by cloudProvider`

Reply:
Thanks for the comment. But the idea is to not capture these in the NodeRegistrationHealthy status condition. If the NodePool isn't referencing the supported nodeClass then the NodePool itself will not go ready.